proposals/004-skips/README.md

```diff
@@ -0,0 +1,179 @@
+# SKIP 000 -Sidecar Containers in Spin Operator
+
+Summary: This proposal introduces sidecar container support in the Spin Operator, enabling Spin applications to run alongside Linux containers within the same pod. This allows for added functionality such as monitoring, logging, and security.
+
+Owner: jiazh@microsoft.com
+
+Impacted Projects:
+
+- [x] spin-operator
+- [ ] `spin kube` plugin
+- [ ] runtime-class-manager
+- [ ] containerd-shim-spin
+- [ ] Governance
+- [ ] Creates a new project
+
+Created: October 16th, 2024
+
+Updated: October 16th, 2024
+
+## Background
+
+The [Runwasi] project supports running Wasm workloads in the same pod as Linux containers. Currently, the SpinApp Custom Resource Definition (CRD) does not support running [Sidecar Containers]. Sidecars are important in use cases where additional services need to run in the same pod as the Spin application, enabling low-latency communication and resource sharing.
+
+[Sidecar Containers]: (https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/)
+[Runwasi]: (https://github.com/containerd/runwasi)
+[SpinApp CRD]: (https://www.spinkube.dev/docs/reference/spin-app/)
+[Spin Operators repository]: (https://github.com/spinkube/spin-operator)
+
+### User Stories
```

Comment:
Is there a use case where sidecars can be another Wasm app?

Reply:
There are use cases for Wasm app as init containers, but not as sidecars AFAIK so far. In theory, Wasm could be a great alternative to Linux sidecar due to it's advantage in memory usage and artifact size.